File: system-test/java/src/main/java/com/optivem/eshop/dsl/port/then/ThenResultStage.java
package com.optivem.eshop.dsl.port.then;

import com.optivem.eshop.dsl.port.then.steps.ThenFailure;
import com.optivem.eshop.dsl.port.then.steps.ThenSuccess;

public interface ThenResultStage extends ThenStage {
    ThenSuccess shouldSucceed();

    ThenFailure shouldFail();
}

File: system-test/typescript/src/testkit/dsl/port/then/then-result-stage.ts
import type { ThenStage } from './then-stage.js';
import type { ThenSuccess } from './steps/then-success.js';
import type { ThenFailure } from './steps/then-failure.js';

export interface ThenResultStage extends ThenStage {
  shouldSucceed(): ThenSuccess;
  shouldFail(): ThenFailure;
}
